prow/istio-postsubmit.sh

```diff
@@ -29,29 +29,11 @@ set -u
 # Print commands
 set -x
 
-if [ "${CI:-}" == 'bootstrap' ]; then
-  export USER=Prow
-
-  # Test harness will checkout code to directory $GOPATH/src/github.com/istio/istio
-  # but we depend on being at path $GOPATH/src/istio.io/istio for imports
-  mv ${GOPATH}/src/github.com/istio ${GOPATH}/src/istio.io
-  ROOT=${GOPATH}/src/istio.io/istio
-  cd ${GOPATH}/src/istio.io/istio
-
-  # Use the provided pull head sha, from prow.
-  GIT_SHA="${PULL_BASE_SHA}"
-
-  # Use volume mount from pilot-presubmit job's pod spec.
-  # FIXME pilot should not need this
-  ln -sf "${HOME}/.kube/config" pilot/pkg/kube/config
-else
-  # Use the current commit.
-  GIT_SHA="$(git rev-parse --verify HEAD)"
-fi
-cd $ROOT
+source ${ROOT}/prow/lib.sh
+setup_and_export_git_sha
 
-# Build
-${ROOT}/bin/init.sh
+cd $ROOT
+make init
```

Comment:
I'm not clear on the current state of dependencies, but there can be a perf benefit to batching together the calls to make.  E.g., the "make init" could be part of the next call to make, such as:
<pre>time make init localTestEnv test</pre>
or possibly even combine all 3 together
<pre>HUB="gcr.io/istio-testing"
TAG="${GIT_SHA}"
time ISTIO_DOCKER_HUB="${HUB}" make init localTestEnv test push HUB="${HUB}" TAG="${TAG}"
</pre>

This is more an observation than a request, and I'm not sure if there's be much of a benefit now.  I had opted for robust depedency checking that's likely favor this kind of approach, but other changes were made to take shortcuts to benefit the split builds in CircleCI and I'm not sure what the end result is like.